### PR TITLE
Fix/1099/unfocus does not respect the focus depth

### PR DESCRIPTION
**Pull request by DmytroHutkin**
# {Unfocus does not respect the focus depth}

_**Please read the [CONTRIBUTING.md](https://github.com/MaibornWolff/codecharta/blob/main/CONTRIBUTING.md) before opening a PR.**_

Closes: #1099 

## Description

**While unfocusing a building/folder it is now possible to focus on a previously focused node if a focus hierarchy exists**

**Three buttons depending on situation available**:
  - Unfocus node
  - Unfocus parent (on a building) to unfocuse it's folder
  - Unfocus all (to show complete map) if there are more than one focused node

## Screenshots or gifs
**BEFORE:**

before
**AFTER:**

after

### CODE REVIEW

File: visualization/app/codeCharta/ui/fileChooser/fileChooser.component.ts [hunk 1]
```diff
@@ -16,6 +17,7 @@ export class FileChooserController {
 	constructor(private $scope, private codeChartaService: CodeChartaService, private storeService: StoreService) {}
 
 	onImportNewFiles(element) {
+		UnfocusButtonController.NEW_FILES_LOADED = true
```
Comment (ce-bo) [inline visualization/app/codeCharta/ui/fileChooser/fileChooser.component.ts:20]: Wasn't this possible with listening to en event?

File: visualization/app/codeCharta/ui/unfocusButton/unfocusButton.component.spec.ts [hunk 2]
```diff
@@ -71,14 +71,51 @@ describe("UnfocusButtonController", () => {
...
-			storeService.dispatch(focusNode("/root"))
+		it("should apply previous focusedNodePath in storeService", () => {
+			unfocusButtonController["_viewModel"].focusedNodes = ["/root", "/root/app"]
+			storeService.dispatch(focusNode("/root/app"))
 
 			unfocusButtonController.removeFocusedNode()
 
+			expect(unfocusButtonController["_viewModel"].focusedNodes.length).toBe(1)
+			expect(unfocusButtonController["_viewModel"].focusedNodes).toEqual(["/root"])
+			//expect(storeService.getState().dynamicSettings.focusedNodePath).toEqual("/root")
```
Comment (ce-bo) [inline visualization/app/codeCharta/ui/unfocusButton/unfocusButton.component.spec.ts:109]: Can we remove this line?

Comment (DmytroHutkin) [reply visualization/app/codeCharta/ui/unfocusButton/unfocusButton.component.spec.ts:109]: Yes

File: visualization/app/codeCharta/ui/fileChooser/fileChooser.component.ts (hunk 1 above)
Comment (DmytroHutkin) [reply visualization/app/codeCharta/ui/fileChooser/fileChooser.component.ts:20]: I actually don't know)

Comment (DmytroHutkin) [reply visualization/app/codeCharta/ui/fileChooser/fileChooser.component.ts]: Well, it was pretty simple and it's done now :)

**Review by ce-bo — APPROVED** [review]
LGTM :)

**Pull request merged**